### DISCUSSION TITLE
Role based permissions

### DISCUSSION THREAD

**Discussion by govind4u** [NationalSecurityAgency/skills-service#827] [Q&A] [answered]
As part of OKTA integration with SkillTree, I was creating/updating the users with **_ROLE_APP_USER_** role by default (on successful okta login) .  When the user logs in the project page is showing Create Project button.  I would want a normal user with _**ROLE_APP_USER**_ role should have access to progress and ranking views only.   Could you please elaborate the permissions assigned for the below roles?  

- ROLE_APP_USER
- ROLE_PROJECT_ADMIN
- ROLE_SUPERVISOR
- ROLE_SUPER_DUPER_USER

**Comment by sudo-may** [comment, marked as answer]
@govind4u Thank you for your interest in SkillTree! The most common use-case is to use the SkillTree dashboard for administrating a gamificaiton profile and then use the SkillTree  [client libraries](https://code.nsa.gov/skills-docs/skills-client/#skilltree-client-libraries) to embed user's ranking and visualization display into your application. SkillTree  [client libraries](https://code.nsa.gov/skills-docs/skills-client/#skilltree-client-libraries) visualize data for the currently logged-in user, these libraries provide fully navigable and themeable Ranking and Visualization views that implement display for points, levels, badges, progress, levels, etc.. Detailed integration guide can be found [here](https://code.nsa.gov/skills-docs/skills-client/#skilltree-client-libraries)..

If this is not your use-case then,

The SkillTree dashboard also has the concept of [Progress and Ranking](https://code.nsa.gov/skills-docs/dashboard/user-guide/progress-and-ranking.html) pages which visualize the user's current Progress and Ranking across multiple projects. These views can be enabled via these SkillTree [installation properties](https://code.nsa.gov/skills-docs/dashboard/install-guide/config.html#progress-and-ranking-views). I am guessing these views are referenced in the question. SkillTree currently does not implement a role that would provide access to these [Progress and Ranking](https://code.nsa.gov/skills-docs/dashboard/user-guide/progress-and-ranking.html) pages within the dashboard while disallowing access to admin pages. However, there is a property to change the default landing page: 
```
# optionally change default for the landing page (admin is the default)
skills.config.ui.defaultLandingPage=progress
```

Although any user can access admin pages, a user can only view and administer a project that he/she created or for which the user has been granted admin permission (ROLE_PROJECT_ADMIN). 

Please let us know if this makes sense and please feel free to let us know more about your use-case/usage.

**Comment by sudo-may** [comment]
Hello @govind4u,

[skills-service:2.0](https://skilltreeplatform.dev/release-notes/skills-service.html#_2-0-0) introduced the [Private Invite Only Project](https://skilltreeplatform.dev/dashboard/user-guide/projects.html#setting-visibility) feature - Progress and Ranking pages can only be accessed by users who have been invited to join the project and who have accepted the invite, any other user attempting to access the project will receive an Access Denied error.
